applications/RomApplication/costum_strategies/rom_builder_and_solver.h

```diff
@@ -392,6 +385,9 @@ class ROMBuilderAndSolver : public BuilderAndSolver<TSparseSpace, TDenseSpace, T
             bool condition_is_active = true;
             if ((it)->IsDefined(ACTIVE))
                 condition_is_active = (it)->Is(ACTIVE);
+            //detect if the condition has a Hyperreduced Weight (H-ROM simulation) or not (ROM simulation)
+            if (!((it)->Has(HROM_WEIGHT)))

```

Comment:
I would do this checking in the Initialize so that it is done only once in the whole simulation